### PR TITLE
DCOS-38119: Typescript type definitions

### PR DESCRIPTION
**Pull request by nLight**
With these definitions my stuff compiles and works \o/ Feel free reviewing it / copy-pasting in your branches.

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

Closes DCOS-38119

### CODE REVIEW

**Comment by juliangieseke** [comment]
Thanks a lot! IMHO we can merge this as is (almost, commit message needs to be fixed)

File: src/js/components/ToggleButton.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+interface IToggleButtonProps {
+  checked: boolean;
+  children: any;
+  onChange: () => void;
+  checkboxClassName: string[] | object | string;
+  className: string[] | object | string;
+}
+
+export default class ToggleButton extends Component<IToggleButtonProps, {}> {}
```
Comment (juliangieseke) [inline src/js/components/ToggleButton.d.ts]: AFAIK you can say `Component<PropsInterface>` and dont need to pass a state type if there is no state - and if there is state, I think we should add it 🤷‍♂️

File: src/js/components/modals/ModalHeading.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { ReactElement } from "react";
+
+type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;
```
Comment (juliangieseke) [inline src/js/components/modals/ModalHeading.d.ts]: this translates to string, correct? (in js component)

File: src/typings/react-ace.d.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Component, ReactElement } from "react";
+
+export default class Ace extends Component {
+  props: {
```
Comment (juliangieseke) [inline src/typings/react-ace.d.ts]: After fiddling around with some types: I like the `Component<Props>` approach way more 👍

File: src/js/components/modals/ModalHeading.d.ts (hunk 2 above)
Comment (nLight) [reply src/js/components/modals/ModalHeading.d.ts]: According to propTypes those should be numbers. Then they translate to strings

**Comment by nLight** [comment]
Everything should be fixed now @Fabs @juliangieseke

**Comment by nLight** [comment]
@juliangieseke sorry :D I added ts configs

**Comment by juliangieseke** [comment]
…now you need to change the commit message :)

**Comment by nLight** [comment]
@juliangieseke doh! Fixed

File: src/js/structs/Item.d.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default class Item {
```
Comment (Fabs) [inline src/js/structs/Item.d.ts]: This starts with I :trollface:

Comment (nLight) [reply src/js/structs/Item.d.ts]: it's not an interface (touché) :D

**Comment by nLight** [comment]
@juliangieseke @Fabs github made me push force

**Pull request merged**